index-server/entrypoint.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+# turn on bash's job control
+set -m
+
+# Check if devfile stacks and index.json exist
+if [ ! -d "$DEVFILE_STACKS" ]; then
+    echo "The container does not contain any devfile stacks in $DEVFILE_STACKS. Exiting..."
+    exit 1
+fi
+if [ ! -e "$DEVFILE_INDEX" ]; then
+    echo "The container does not contain an index.json at $DEVFILE_INDEX. Exiting..."
+    exit 1
+fi
+
+# Start the nginx server
+nginx -g "daemon off;" &
+
+# Start the index server
+/registry/index-server
```

Comment:
Curious, why do we start the nginx proxy first, rather than the index server? My thinking was something like this:
```
# Start the index server
/registry/index-server &

# Start the nginx server
exec "$@"
```

(Both should work fine, just curious)

Reply:
This is interesting, actually I had the same thought as yours and tried the code block above (starts index server process firstly at backgeound), but it failed, I went into the index container for deeper debugging, for some reasons the index server process was not in the index container and I can only saw the nginx server process. I couldn't figure out the reasons but probably it's due to docker doesn't have a really good support for the case where multiple services in one container, especially multiple services need communication. Probably entrypoint process must start firstly? 🤔

Reply:
`exec` command uses the existing bash process to run the command instead of creating new process and docker uses it as entrypoint process, from docker perspective, I am not sure if that results in another background process that we trigger in the same script gets ignored.

Reply:
Huh, interesting. In that case, let's stick with that you're doing.